src/Http/Firewall.php

```diff
@@ -0,0 +1,177 @@
+     *       on endpoints that do not require a central access.
+     */
+    private const STRATEGY_DEFAULT_FOR_CORE = self::STRATEGY_AUTHENTICATED;
+
+    /**
+     * Security strategy to apply by default on plugin ajax/front scripts.
+     *
+     * @TODO In GLPI 10.1, raise default level to `self::STRATEGY_AUTHENTICATED`.
+     */
+    private const STRATEGY_DEFAULT_FOR_PLUGINS = self::STRATEGY_NO_CHECK;
+
+    /**
+     * GLPI root directory.
+     * @var string
+     */
+    private string $root_dir;
+
+    /**
+     * GLPI plugins root directories.
+     * @var string[]
+     */
+    private array $plugins_dirs;
+
+    public function __construct(string $root_dir = GLPI_ROOT, array $plugins_dirs = PLUGINS_DIRECTORIES)
+    {
+        $this->root_dir = $root_dir;
+        $this->plugins_dirs = $plugins_dirs;
+    }
+
+    /**
+     * Apply the firewall strategy for given path.
+     * If no strategy is provided, the corresponding default strategy will be applied.
+     */
+    public function applyStrategy(string $path, ?string $strategy = null): void
+    {
+        if ($strategy === null) {
+            $strategy = $this->computeDefaultStrategy($path);
+        }
+
+        switch ($strategy) {
+            case self::STRATEGY_AUTHENTICATED:
+                Session::checkLoginUser();
+                break;
+            case self::STRATEGY_CENTRAL_ACCESS:
+                Session::checkCentralAccess();
+                break;
+            case self::STRATEGY_HELPDESK_ACCESS:
+                Session::checkHelpdeskAccess();
+                break;
+            case self::STRATEGY_FAQ_ACCESS:
+                Session::checkFaqAccess();
+                break;
+            case self::STRATEGY_NO_CHECK:
+                // nothing to do
+                break;
+            default:
+                trigger_error(sprintf('Invalid `%s` strategy.', $strategy), E_USER_WARNING);
+                break;
+        }
+    }
+
+    private function computeDefaultStrategy(string $path): string
+    {
+        // Check if entrypoint is a GLPI core ajax/front script.
+        if (
+            str_starts_with($path, '/ajax/')
+            || str_starts_with($path, '/front/')
+        ) {
+            return self::STRATEGY_DEFAULT_FOR_CORE;
+        }
+
+        // Check if entrypoint is a plugin ajax/front script.
+        foreach ($this->plugins_dirs as $plugins_dir) {
+            $relative_path = preg_replace(
+                '/^' . preg_quote($this->normalizePath($this->root_dir), '/') . '/',
+                '',
+                $this->normalizePath($plugins_dir)
+            );
+
+            if (preg_match('/^' . preg_quote($relative_path, '/') . '\/[^\/]+\/(ajax|front)\/' . '/', $path) === 1) {
+                // Entrypoint is a plugin ajax/front script.
+                return self::STRATEGY_DEFAULT_FOR_PLUGINS;
+            }
```

Comment:
This condition seems to never true, even when `$path` refer to a plugin.

For example:
* `$path` will be `/glpi/10.0-bugfixes/plugins/formcreator/front/form.php`  
* `$relative_path` will be `/plugins`

The condition look for a string starting with `$relative_path`, which is not the case here.

![image](https://github.com/glpi-project/glpi/assets/42734840/8f8a1da0-da59-4a42-848e-4176e5dd14f2)

Note that removing `^` from the regex find the correct results, but I have no idea if this is the solution to follow (maybe you expected `$path` to start with `/plugins` and its value should be fixed instead).

![image](https://github.com/glpi-project/glpi/assets/42734840/5dde610b-25bc-47f0-85c7-3ce0c80f84ce)

This failure is not detected by the units tests because the default strategy for plugins is also the fallback value of `computeDefaultStrategy` (so we fallback to the right value by mistake).

To improve tests quality, plugin/code detection should be splitted in another unit tested function who return the entry point type (core, plugin or other).

Reply:
GLPI URL prefix was not supported. It is now fixed.

My first draft was using `central_access` for GLPI core and `authenticated` for plugins, and this is why this code exists. It was too restrictive and a BC-break for plugins. I decided to remove the default strategy for plugins, and, maybe, to reactivate it in GLPI 10.1. As the code exists, I decided to keep it here.